### PR TITLE
#496 Target for MatButton / MatIconButton

### PR DESCRIPTION
**Pull request by enkodellc**
I added a Target Property for MatButton and MatIconButton.   I tried to this for the MatNavItem as well but it would open a new page but also redirect the current page. I commented out the MatNavItem since the other two worked great. Please review this PR and Merge if you think this is the correct approach.

### CODE REVIEW

**Comment by lindespang** [comment]
@enkodellc Hi Keith! Nice work.

There is already a JS runtime injected in the matblazor base class `BaseMatComponent` which exposes interop to it's children `protected async Task<T> JsInvokeAsync<T>(string code, params object[] args)`

So I think you should call this one instead (Seems to be the case for other JS interops in MatBlazor) to avoid JS inject clutter in children components.

**Comment by enkodellc** [comment]
@lindespang  thanks I will refactor

**Comment by enkodellc** [comment]
@lindespang I updated the JsInvoke. Can you take peak at the Target for MatNavItem... it doesn't work like MatButton or MatIconButton. I cannot figure out why.

**Comment by lindespang** [comment]
@enkodellc Sure hopefully I have time this evening otherwise tomorrow

File: src/MatBlazor/Components/MatButton/BaseMatButton.cs [hunk 1]
```diff
@@ -72,6 +71,12 @@ public BaseMatButton()
         [Parameter]
         public string Link { get; set; }
 
+        /// <summary>
+        /// Target of Link when clicked.
+        /// </summary>
+        [Parameter]
+        public string Target { get; set; } = null;
```
Comment (JayArrowz) [inline src/MatBlazor/Components/MatButton/BaseMatButton.cs:78]:
```suggestion
        public string Target { get; set; };
```

**Comment by lindespang** [comment]
@enkodellc Hi took a look now, the MatNavItem behaves that way because of the asp.net component in the `<li>` gets a href parameter.

**MatNavItem.razor**
image

When I ran this code a few times (clicked a few Target buttons)  i noticed I got TaskCancelled-exceptions since we didn't get response back from the awaited js interop. Not sure why... Thought we would get reference to window back, maybe you know more about `window.open()` ? But if it's a void JS function we need to use ` JSRuntime.InvokeVoidAsync`

**Comment by enkodellc** [comment]
@lindespang I needed to look at the actual code a bit and realized the NavLink can take a target attribute and I do not need to the JsInvoke. Pushing the code now.

**Comment by enkodellc** [comment]
@lindespang please review this PR should resolve issues 492, 496, 218.

**Comment by lindespang** [comment]
@enkodellc Looks good, thanks for doing this! Merged!

Just a little friendly "slap on the wrist" that this PR is polluted with the MatTable changes 😉. You might want to edit title/desc so it's easier for Vladmir to understand when merging to Master.

**Comment by enkodellc** [comment]
I know it wasn't the best approach but at least I labeled by commits for each PR :)

**Pull request merged**